internal/datastore/datastore_test.go

```diff
@@ -2,83 +2,332 @@ package datastore
 
 import (
 	"context"
-	"database/sql"
 	"fmt"
-	"math/rand"
-	"strconv"
 	"sync"
-	"time"
+	"testing"
 
+	"github.com/DATA-DOG/go-sqlmock"
 	"github.com/jinzhu/gorm"
-	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
+	"github.com/nedrocks/delphisbe/internal/config"
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/assert"
 )
 
 type testBackend struct {
 	db              Datastore
 	discussionMutex sync.Mutex
 }
 
-func MakeDatastore(ctx context.Context, testData TestData) (Datastore, func() error, error) {
-	url := "postgres://chatham_local@localhost:5432/"
+func TestDelphisDB_InitializeStatements(t *testing.T) {
+	ctx := context.Background()
 
-	dbName, err := createTestDatabase(ctx, url+"?sslmode=disable")
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to create test database")
-	}
+	Convey("InitializeStatments", t, func() {
+		db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
 
-	url = url + dbName + "?sslmode=disable"
+		assert.Nil(t, err, "Failed setting up sqlmock db")
 
-	db, err := connectTestDatabase(ctx, url)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to create testing datastore")
-	}
+		gormDB, _ := gorm.Open("postgres", db)
+		mockDatastore := &delphisDB{
+			dbConfig:  config.TablesConfig{},
+			sql:       gormDB,
+			pg:        db,
+			prepStmts: &dbPrepStmts{},
+			dynamo:    nil,
+			encoder:   nil,
+		}
+		defer db.Close()
 
-	// Create test tables with test data
-	closeFunc, err := db.CreateTestTables(ctx, testData)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to create test tables")
-	}
+		Convey("when prepared statement fails to initialize", func() {
+			mock.ExpectPrepare(getPostByIDString)
+			mock.ExpectPrepare(getPostsByDiscussionIDFromCursorString).WillReturnError(fmt.Errorf("error"))
+			err := mockDatastore.initializeStatements(ctx)
 
-	return db, closeFunc, nil
-}
+			So(err, ShouldNotBeNil)
+		})
 
-func connectTestDatabase(ctx context.Context, url string) (Datastore, error) {
-	// Initialize gorm
-	testGorm, err := gorm.Open("postgres", url)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to open testing database - gorm")
-	}
-
-	// Initialize sql
-	testDB, err := sql.Open("postgres", url)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to open testing database - SQL")
-	}
-
-	// Create db object
-	return &delphisDB{
-		sql:       testGorm,
-		pg:        testDB,
-		prepStmts: &dbPrepStmts{},
-	}, nil
-}
+		tests := []struct {
```

Comment:
How do we keep this list up to date? There are a few options I can think of -- https://www.nedrockson.com/posts/engineering/testing-with-go-ast/ could be a good breadcrumb here?